kiali.go

```diff
@@ -203,16 +153,8 @@ func validateConfig() error {
 	// log some messages to let the administrator know when credentials are configured certain ways
 	auth := config.Get().Auth
 	log.Infof("Using authentication strategy [%v]", auth.Strategy)
-	if auth.Strategy == config.AuthStrategyLogin {
-		creds := config.Get().Server.Credentials
-		if creds.Username == "" && creds.Passphrase == "" {
-			// This won't cause Kiali to abort, but users won't be able to log in, so immediately log a warning
-			log.Warningf("Credentials are missing. Create a proper secret. Please refer to the documentation for more details.")
-		}
-	} else if auth.Strategy == config.AuthStrategyAnonymous {
+	if auth.Strategy == config.AuthStrategyAnonymous {
```

Comment:
I think this should do more validation.
Manually changing the configmap to use "login" leads to a running pod. The error in config is evident only after trying to login; this is when Kiali rejects the request with error and logs the misconfiguration. 

I think here we should check for a valid strategy and return an error to drive the pod to an failed state. Otherwise, the UI behavior can be undefined.

Reply:
done.

I left the hack scripts alone - those are for older versions that still support login strategy. At some point we'll want to remove those, though - the cluster-openshift.sh is really old and should never be used anyway.